addons/project/models/project_task.py

```diff
@@ -1392,6 +1393,37 @@ def _task_message_auto_subscribe_notify(self, users_per_task):
                     mail_auto_delete=False,
                 )
 
+    def _task_message_unsubscribe_notify(self, unassigned_users_per_task):
+        if self.env.context.get('mail_auto_subscribe_no_notify'):
+            return
+        template_id = self.env['ir.model.data']._xmlid_to_res_id('project.project_message_user_unassigned', raise_if_not_found=False)
+        if not template_id:
+            return
+        task_model_description = self.env['ir.model']._get(self._name).display_name
+        subtype_id_unassigned = self.env['ir.model.data']._xmlid_to_res_id('project.mt_task_unassigned', raise_if_not_found=False)
+        for task, unassigned_users in unassigned_users_per_task.items():
+            followers_with_subtype = task.message_follower_ids.filtered(lambda follower: subtype_id_unassigned in follower.subtype_ids.ids)
+            if not unassigned_users or not followers_with_subtype:
+                continue
+            unassigned_user_names = ', '.join(unassigned_users.mapped('name'))
+            values = {
+                'object': task,
+                'model_description': task_model_description,
+                'access_link': task._notify_get_action_link('view'),
+                'unassigned_user_names': unassigned_user_names,
+            }
+            unassignation_msg = self.env['ir.qweb']._render('project.project_message_user_unassigned', values, minimal_qcontext=True)
+            unassignation_msg = self.env['mail.render.mixin']._replace_local_links(unassignation_msg)
+            task.message_notify(
+                subject=_("%(unassignee_name)s have been unassigned from %(task_name)s", unassignee_name=unassigned_user_names, task_name=task.display_name),
```

Comment:
I meant to literally say "Users have been unassigned..." without citing them in the title to circumvent grammatical number issues. But on the other hand, having the names in the title is also good, so I'll let Alexandra judge